experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -257,3 +257,97 @@ function isomorphism(::Type{PermGroup}, W::WeylGroup; set_properties::Bool=true)
 
   return MapFromFunc(W, G, iso, isoinv)
 end
+
+@doc raw"""
+    map_word(w::WeylGroupElem, genimgs::Vector; init = nothing)
+
+If `init` is `nothing` and `word(w) = [UInt(`$i_1$`), ..., UInt(`$i_n$`)]`, then return the product $R_1 R_2 \cdots R_n$
+with $R_j =$ `genimgs[`$i_j$`]`.
+Otherwise return the product $xR_1 R_2 \cdots R_n$ where $x =$ `init`.
+
+The length of `genimgs` must be equal to the rank of the parent of `w`.
+In particular, it must not be empty.
+If `w` is the trivial element, then `init` is returned if it is different
+from `nothing`, and otherwise `one(genimgs[1])` is returned.
+
+See also: [`map_word(::Union{FPGroupElem, SubFPGroupElem}, ::Vector)`](@ref),
+[`map_word(::Union{PcGroupElem, SubPcGroupElem}, ::Vector)`](@ref).
+Note that `map_word(::WeylGroupElem)` does not provide a `genimgs_inv` parameter
+because the generators of a Weyl group are always self-inverse.
+
+# Examples
+```jldoctest
+julia> W = weyl_group(:B, 3); imgs = [2, 3, 5];
+
+julia> map_word(one(W), imgs)
+1
+
+julia> map_word(W([1]), imgs)
+2
+
+julia> map_word(W([1]), imgs, init=7)
+14
+
+julia> map_word(W([1,2,1]), imgs)
+12
+
+julia> map_word(W([2,1,2]), imgs) # W([2,1,2]) == W([1,2,1])
+12
+
+julia> map_word(W([3, 2, 1, 3, 2, 3]), imgs)
+2250
+```
+"""
+function map_word(w::WeylGroupElem, genimgs::Vector; init=nothing)
+  @req length(genimgs) == number_of_generators(parent(w)) "Length of vector of images does not equal rank of Weyl group"
+  return Oscar.map_word([Int(i) for i in word(w)], genimgs, init=init)
+end
+
+@doc raw"""
+    parabolic_subgroup(W::WeylGroup, vec::Vector{<:Integer}, w::WeylGroupElem) -> WeylGroup, Map{WeylGroup, WeylGroup}
+
+Returns two objects: The subgroup `U` of `W` generated by `[inv(w)*u*w for u in gens(W)[vec]]`, whose generators are set to be this vector and which is itself a `WeylGroup`, and the embedding of `U` into `W`. The elements of `vec` must be pairwise distinct integers in `1:number_of_generators(W)` and `vec` must be non-empty.
```

Comment:
Please wrap lines at something between 80 and 100 columns wide (it helps with reviewing changes made to this text in the future, as the diffs are nicer)

Reply:
- it should be "Return" not "Returns" (see point 2 at <https://docs.julialang.org/en/v1/manual/documentation/>)
- I'd say the return group is *not* that subgroup `U`, rather it is a free standing Weyl group isomorphic to `U`, and return map is an isomorphism between it and `U`.

How about this:
```suggestion
Return a Weyl group `P` and an embedding $f:P\to W$ such that $f(P)$ is
the subgroup `U` of `W` generated by `[inv(w)*u*w for u in gens(W)[vec]]`.
The elements of `vec` must be pairwise distinct integers in
`1:number_of_generators(W)` and `vec` must be non-empty.
```



Oh and an example in form of `jldoctest` would be super nice :-)

Reply:
Probably should also say that `f` maps the i-th generator of P t the i-th generator of `U` or so (at least I assume that's what it does?)

Reply:
Done. The statement that $f(P)$ is generated by `[inv(w)*u*w for u in gens(W)[vec]]` is now technically redundant because it follows from the description of `f` on the generators, but I left it in because it seems more clear to me that way.